### ISSUE TITLE
[1.0] Red screen errors point to commons.js

### ISSUE THREAD

**Issue by 0x80** [gatsbyjs/gatsby#921]
I'm seeing a lot of these kind or errors during development:

```
TypeError: Cannot read property 'allWorksJson' of undefined
Index.render
http://localhost:8000/commons.js:2852:45
Index.render
http://localhost:8000/commons.js:54355:31
http://localhost:8000/commons.js:41627:22
measureLifeCyclePerf
http://localhost:8000/commons.js:40906:13
ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext
http://localhost:8000/commons.js:41626:26
ReactCompositeComponentWrapper._renderValidatedComponent
http://localhost:8000/commons.js:41653:33
ReactCompositeComponentWrapper.performInitialMount
http://localhost:8000/commons.js:41193:31
ReactCompositeComponentWrapper.mountComponent
http://localhost:8000/commons.js:41089:22
Object.mountComponent
http://localhost:8000/commons.js:47603:36
ReactCompositeComponentWrapper.performInitialMount
http://localhost:8000/commons.js:41202:35
```

They make it sometimes hard to figure out what's going on, because they don't point to the original source file. I don't quite understand why they point to commons, because source maps are enabled. Is it maybe because the webpack common chunks extractor is more of a production level step? Do we need this commons file in development mode, or can we maybe disable this chunk?

**Comment by jquense** [comment]
I think this is the case because the red box component doesn't have access to the stack trace  with sourcemaps only the "real" one related to the actual files

**Comment by 0x80** [comment]
@jquense Is this different for gatsby compared to normal webpack development? I've use it before where it was displaying useful stack traces, so I'm wondering if it can be improved.

**Comment by KyleAMathews** [comment]
TBH, I'm a bit confused about this too. I don't remember it doing this before so perhaps something with our config broke.

I've been intending to do a deep-dive on create-react-app's way of handling webpack errors and porting that over. Probably in conjunction with upgrade Gatsby to webpack 2. Lemme know if any of you want to work on that first. Have some other things to get to first.

**Comment by 0x80** [comment]
I did some more digging and read in the redbox-react docs that they recommend setting `devtool` to `eval`. This solved it for me. The redbox errors are now pointing to the right files:

```
ReferenceError: Bam is not defined
Index.render
webpack:///./src/pages/index.js?:99:41
Index.render
webpack:///./~/react-proxy/modules/createPrototypeProxy.js?:44:30
eval
webpack:///./~/react-dom/lib/ReactCompositeComponent.js?:796:21
```
Not sure if using eval has other downsides. Maybe sacrificing some build speed. I can't seem to remember all these webpack devtool options and their pro/cons. I feel like I've tried them all in the past 2 years and it's still trail and error 😒
